pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Consumer.java

```diff
@@ -536,19 +537,38 @@ CompletableFuture<Void> reconsumeLaterCumulativeAsync(Message<?> message,
     CompletableFuture<Void> seekAsync(long timestamp);
 
     /**
-     * Get the last message id available for consume.
+     * Get the last message id of the topic subscribed.
      *
-     * @return the last message id.
+     * @return the last message id of the topic subscribed
+     * @throws PulsarClientException if multiple topics or partitioned topics are subscribed or failed because of a
+     *   network issue
+     * NOTE: Use {@link Consumer#getLastMessageIds()} instead.
      */
+    @Deprecated
```

Comment:
If we mark the API as deprecated, we should not change the behavior of the API. The `@Deprecated` annotation helped us to tell users the API will be removed in the future, but not now. Users can migrate to the new API after upgrading to the new version without any compatibility issues. We should mark it as `@Deprecated` first and remove it in the future.

Reply:
I restored the behaviour now. And the `MultiMessageIdImpl` related tests are still removed.

Reply:
I think we should keep the test until we remove the API. Otherwise, we might introduce breaking changes to the deprecated API in the subsequent releases, maybe a patch release of the LTS version. The deprecated API should also be under the guarantee until it is removed.

Reply:
It makes sense to me. Let's remove them when we remove the whole `getLastMessageId` API.